lib/model.js

```diff
@@ -484,6 +484,7 @@ var validateIncludedElements = function(options, tableNames) {
         include.subQuery = false;
       } else {
         include.subQueryFilter = false;
+        include.subQuery = include.subQuery || include.hasParentRequired && (include.hasDuplicating || include.hasRequired);
```

Comment:
I just need the `true` value here because its causing the nested queries to go with LEFT OUTER JOIN even if they should be part of INNER JOIN.

If we consider we just need to respect `hasParentRequired` and `hasRequired` flags only, so there is no particular role for `hasDuplicating` here. I think I saw some test failing because of that. But that time code didn't included the `hasParentRequired` which I added later on after your suggestion. I will repeat the tests to check if we really need the `hasDuplicating` now.